sky/backends/backend_utils.py

```diff
@@ -1786,21 +1786,40 @@ def _update_cluster_status_no_lock(
             if returncode:
                 raise exceptions.FetchIPError(
                     reason=exceptions.FetchIPError.Reason.HEAD)
+        ray_cluster_up = True
+        # For non-spot clusters:
         # If we get node ips correctly, the cluster is UP. It is safe to
         # set the status to UP, as the `handle.external_ips` function uses ray
         # to fetch IPs and starting ray is the final step of sky launch.
```

Comment:
Actually, it may not be true for multi-node either, as the `ray get-node-ips` only use the tag on the cloud without actually checking the ray cluster status. We may want to make L1778 check status for all the nodes instead.

Reply:
Ah I see. I remove the `if` and now it also run `ray cluster` for multi-node case. just want to make sure that's what we want right?

Reply:
I was thinking about something similar as the following code to check the number of workers is the same as the expected number with `ray status`.

We can refactor this part of code out and use it?
https://github.com/skypilot-org/skypilot/blob/8628a254892114fdc4068d234976de731612c4b8/sky/backends/backend_utils.py#L1033-L1078

Reply:
Ahh got it. Parsing the output from `ray status` is needed. I just updated the code. PTAL.